classes/chat/ChatManager.php

```diff
@@ -142,6 +150,45 @@ private function fetchPosts(?int $starting_post_id = null, int $max_posts = self
             }
             $post->message = nl2br($this->system->html_parse($post->message, false, true));
 
+            // Handle Quotes
+            $pattern = "/\[quote:\d+\]/";
+            $has_quote = preg_match_all($pattern, $post->message, $matches);
+            if ($has_quote) {
+                if ($allow_quote) {
+                    foreach ($matches[0] as $match) {
+                        // get chat post contents
+                        $pattern = "/\[quote:(\d+)\]/";
+                        preg_match($pattern, $match, $id);
+                        $quote = $this->fetchPosts($id[1], 1, false);
+                        if ($quote) {
+                            // if id match
+                            if ($quote[0]->id == $id[1]) {
+                                // format each entry in $quotes
+                                $formatted_quote = "<div class='quote_container'><a class='chat_user_name " . implode(' ', $quote[0]->user_link_class_names) . "' href='" . $this->system->router->getURL("members", ["user" => $quote[0]->user_name]) . "'>" . $quote[0]->user_name . "</a><div class='quote_message'>" . $quote[0]->message . "</div></div>";
+                                // run replace on $test using $matches as search and $quotes as replace
+                                $post->message = str_replace($match, $formatted_quote, $post->message);
+                            }
+                            else {
+                                $post->message = str_replace($matches[0], "<i>(removed)</i>", $post->message);
+                            }
+                        }
+                    }
+                }
+                else {
+                    $post->message = str_replace($matches[0], "(...)", $post->message);
+                }
+            }
+
+            // Handle Mention
+            $pattern = "/@([^ \n\s]+)(?=[^A-Za-z0-9_]|$)/";
```

Comment:
Have you tested which cases stop the username? I'm not sure if it's a regex101 problem but it suggests that non-word characters in the middle of the string would be included
![image](https://github.com/elementum-games/shinobi-chronicles/assets/1691258/5ddc4947-363e-4a6e-b471-b5d4c99c442f)
![image](https://github.com/elementum-games/shinobi-chronicles/assets/1691258/5a9185d2-607d-4cb4-aef0-6cd0b1462e82)

Reply:
Fixed in commit.